usaspending_api/accounts/v2/filters/award_financial.py

```diff
@@ -41,24 +40,31 @@ def award_financial_filter(filters):
 
     queryset = FinancialAccountsByAwards.objects
 
-    # TODO: Derivations
-    # queryset = queryset.annotate(
-    #     treasury_account_symbol=Concat(
-    #         'treasury_account__agency_id',
-    #         Value('-'),
-    #         Case(When(treasury_account__availability_type_code='X', then=Value('X')),
-    #              default=Concat('treasury_account__beginning_period_of_availability', Value('/'),
-    #                             'treasury_account__ending_period_of_availability'),
-    #              output_field=CharField()),
-    #         Value('-'),
-    #         'treasury_account__main_account_code',
-    #         Value('-'),
-    #         'treasury_account__sub_account_code',
-    #         output_field=CharField()),
-    #     recipient_parent_name=Case(
-    #         When(award__latest_transaction__type__in=list(contract_type_mapping.keys()),
-    #              then='award__latest_transaction__contract_data__ultimate_parent_legal_enti'),
-    #         default='award__latest_transaction__assistance_data__ultimate_parent_legal_enti',
-    #         output_field=CharField()))
+    # Derive treasury_account_symbol, allocation_transfer_agency_name, agency_name, federal_account_symbol,
+    # and recipient_parent_name
+    ata_subquery = ToptierAgency.objects.filter(cgac_code=OuterRef('treasury_account__allocation_transfer_agency_id'))
+    agency_name_subquery = ToptierAgency.objects.filter(cgac_code=OuterRef('treasury_account__agency_id'))
+    queryset = queryset.annotate(
+        treasury_account_symbol=Concat(
+            'treasury_account__agency_id',
+            Value('-'),
+            Case(When(treasury_account__availability_type_code='X', then=Value('X')),
+                 default=Concat('treasury_account__beginning_period_of_availability', Value('/'),
+                                'treasury_account__ending_period_of_availability'),
+                 output_field=CharField()),
+            Value('-'),
+            'treasury_account__main_account_code',
```

Comment:
I wonder if there's any way to refactor this cleanly (into a common method? not sure if thats possible given the slight keyword argument variances, but maybe?) since it is so similar to the corresponding code in object_class_program_activity.py (lines 38-58) and account_balances.py (line 43-63)...